src/site/_data/macroOptions.js

```diff
@@ -0,0 +1,16 @@
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const { getAllComponentNames, getMacroOptions } = require('../../lib')
+
+module.exports = async function getAllMacroOptions() {
+  const componentNames = getAllComponentNames()
+  const componentOptions = []
+  // Note that parallelising these causes all components to have the same macro options
+  // Probably caused by https://github.com/microsoft/TypeScript/issues/51554
+  // (fixed in TypeScript 5)
+  // eslint-disable-next-line no-restricted-syntax
+  for (const componentName of componentNames) {
+    // eslint-disable-next-line no-await-in-loop
+    componentOptions.push([componentName, await getMacroOptions(componentName)])
+  }
+  return Object.fromEntries(componentOptions)
+}
```

Comment:
This is a bit slow (~900ms for me). There's a message about it when running `npm run build`. Would be interested in how long it takes for you @adamledwards .